### ISSUE TITLE
Extend CI tests to catch format errors on the web apps

### ISSUE THREAD

**Issue by kimwnasptd** [kubeflow/kubeflow#5810]
/kind feature

**Why you need this feature:**
To ensure the code of our [crud-web-apps](https://github.com/kubeflow/kubeflow/tree/master/components/crud-web-apps) properly formatted.

**Describe the solution you'd like:**
Extend the existing CI python files to also run a task that will be checking the formatting of TypeScript as well as run `flake8` tests on the backends of the [crud-web-apps](https://github.com/kubeflow/kubeflow/tree/master/components/crud-web-apps).

/priority p1
/assign @kimwnasptd

**Comment by DavidSpek** [comment]
@kimwnasptd Something I've noticed is that all the presubmit tests run as a single `presubmit` test. I think it would be better to split the tests so that you can `/retest` just the `common-ui-tests` if it fails for example and not also have it run the volumes-web-app image build test as well. 
This way, you easily see in GitHub that the linting presubmit test failed and that functionally everything is fine, and you can retest only the one that failed as described above.

**Comment by kimwnasptd** [comment]
This would make sense. 

@PatrickXYS if we had multiple presubmit jobs defined in the `prow_config.yaml` and some of them failed, then if we would type the `/retest` would that make only the failed argo workflows rerun?

**Comment by DavidSpek** [comment]
@kimwnasptd `/retest` will retest all, but you would have the option to run `/retest <failed-job>` with that setup. Now you can do `/retest presubmit` which re-runs all the tests as the presubmit contains all the tests.

**Comment by kimwnasptd** [comment]
> Now you can do /retest presubmit which re-runs all the tests as the presubmit contains all the tests.

IIRC right now even if we define multiple presubmits jobs in the `prow_config.yaml` these will end up being run/managed under a single Prow Job, which will spawn multiple Argo CRs for each presubmit job in that file. So if we do `/retest` or `/retest <failed-job>` then this will just rerun the entire failed Prow Job, not just the failing underlying Argo Workflows that are managed from the Prow Job.

@PatrickXYS is my understanding accurate?

**Comment by PatrickXYS** [comment]
> IIRC right now even if we define multiple presubmits jobs in the prow_config.yaml these will end up being run/managed under a single Prow Job, which will spawn multiple Argo CRs for each presubmit job in that file. So if we do /retest or /retest <failed-job> then this will just rerun the entire failed Prow Job, not just the failing underlying Argo Workflows that are managed from the Prow Job.

You're right, in short, to achieve what you desire, you have to add more entries here https://github.com/kubeflow/testing/blob/master/aws/User/clusters/optional-test-infra-prow/namespaces/prow/configmap/config.yaml#L120-L133.

**Comment by kimwnasptd** [comment]
Thanks for the info @PatrickXYS!

Since perfect is the biggest enemy of good enough I propose the following plan:
1. We move forward with just adding extra tasks to the Argo Workflows of our components and do the format checks there for now
2. Investigate in parallel how we could add extra Prow Jobs to be triggered that would run format checks on the codebase after each PR
3. Once we've ensured that our code is formatted we can add global Prow Jobs that will be checking the code format
4. Remove our redundant checks from the components Argo Workflows

**Comment by kimwnasptd** [comment]
/lifecycle frozen

I want to keep this discussion open to see how we could decouple some of the steps that currently run in a single, big, Argo Workflow. This discussion will help us evaluate if it would also make sense to have some GitHub actions for running some simple checks, like formatting.
